source/server/connection_handler_impl.cc

```diff
@@ -15,10 +15,13 @@ namespace Envoy {
 namespace Server {
 
 ConnectionHandlerImpl::ConnectionHandlerImpl(spdlog::logger& logger, Event::Dispatcher& dispatcher)
-    : logger_(logger), dispatcher_(dispatcher) {}
+    : logger_(logger), dispatcher_(dispatcher), disable_listeners_(false) {}
 
 void ConnectionHandlerImpl::addListener(Network::ListenerConfig& config) {
   ActiveListenerPtr l(new ActiveListener(*this, config));
+  if (disable_listeners_) {
+    l->listener_->disable();
```

Comment:
Yeah, with the current implementation we'll start listening in the constructor and then stop here.  I could plumb down a disabled bool parameter (ActiveListener::ActiveListener -> Dispatcher::createListener -> Listener::Listener) which would allow us to close this window and start the listener in a disabled state - evconnlistener_new has a LEV_OPT_DISABLED flag to allow this (http://www.wangafu.net/~nickm/libevent-book/Ref8_listener.html). Do you think that is preferable?

Reply:
I'm not sure it's worth the effort. I think what you have is probably fine. WDYT?

Reply:
Agreed, I think the current approach is good enough.